### PR TITLE
Add timestamp to generated docs

### PR DESCRIPTION
**Pull request by Varunram**
Intended as a solution to #2115.
Contrary to what was proposed back in the issue, appending at the end of the file seemed to be an easier alternative than reading from the file and rewriting - I'm new here, so please correct me if I'm wrong.

### CODE REVIEW

**Comment by Varunram** [comment]
Signed the CLA!

File: doc-tool/src/dotty/tools/dottydoc/staticsite/Site.scala [hunk 1]
```diff
@@ -311,6 +315,16 @@ case class Site(val root: JFile, val projectTitle: String, val documentation: Ma
     * defaults, the user-defined one will take precedence.
     */
   val layouts: Map[String, Layout] = {
+
+    val cal = Calendar.getInstance()
```
Comment (smarter) [inline doc-tool/src/dotty/tools/dottydoc/staticsite/Site.scala]: I would recommend against using java.util.Date/Calendar/SimpleDataFormat. These APIs have many pitfalls and are not thread-safe.  Java 8 introduced a new set of APIs to deal with date and time in the `java.time` package that should be used instead. To get the current time you should use LocalDateTime.now: https://docs.oracle.com/javase/8/docs/api/java/time/LocalDateTime.html#now--

File: doc-tool/src/dotty/tools/dottydoc/staticsite/Site.scala [hunk 2]
```diff
@@ -311,6 +315,16 @@ case class Site(val root: JFile, val projectTitle: String, val documentation: Ma
...
+    val today = cal.getTime()
+    val currentHour = new SimpleDateFormat("hh").format(today)
+    val currentMinute = new SimpleDateFormat("mm").format(today)
+    val amOrPm = new SimpleDateFormat("a").format(today)
+    val x = "<!-- generated on " + cal.get(Calendar.DATE) + "/"+ (cal.get(Calendar.MONTH)+1)+ "/"+cal.get(Calendar.YEAR) + " " + currentHour + ":" + currentMinute + ":" + amOrPm + "-->"
+    var loc = root.getAbsolutePath.toString + "/.." + "/doc-tool/resources/_layouts/main.html"
+    scala.tools.nsc.io.File(loc).appendAll(x)
```
Comment (smarter) [inline doc-tool/src/dotty/tools/dottydoc/staticsite/Site.scala]: This is an alias for dotty.tools.io.File, use that instead.

File: doc-tool/src/dotty/tools/dottydoc/staticsite/Site.scala (hunk 1 above)
Comment (Varunram) [inline doc-tool/src/dotty/tools/dottydoc/staticsite/Site.scala]: Thank you very much! Doing it right away!

File: doc-tool/src/dotty/tools/dottydoc/staticsite/Site.scala (hunk 2 above)
Comment (Varunram) [inline doc-tool/src/dotty/tools/dottydoc/staticsite/Site.scala]: Thanks again and Sorry. Regarding the failing tests, would it be better if I add a check for ```loc ``` over here?

**Comment by Varunram** [comment]
CLA signed already

**Comment by Varunram** [comment]
Noticed this just now, sorry! Would a tacky fix along the lines of ```case class PageInfo(url: String, date: String = (LocalDateTime.now().getDayOfMonth() + "/"+ LocalDateTime.now().getMonthValue()+ "/"+ LocalDateTime.now().getYear() + " " + LocalDateTime.now().getHour() + ":" + LocalDateTime.now().getMinute() + ":" + LocalDateTime.now().getSecond()).toString) {
``` in ```DefaultParams.scala``` be acceptable?

**Comment by felixmulder** [comment]
@Varunram - we want to support people in doing something like:

```
{{ "2017-10-16" | date: '%B %d, %Y' }}
```

Therefore it might be better to fill in the correct date in `DefaultParams.scala` and use that if there wasn't a date already defined in the preamble.

**Comment by Varunram** [comment]
Pushed back irrelevant commits and did the requested changes. Thanks to @felixmulder for helping me out 😄

**Comment by felixmulder** [comment]
LGTM, thanks @Varunram !

**Pull request merged**